### PR TITLE
fix: wrong autoReadDisabledRateLimiting status when limit by maxPendingSendRequests

### LINKED ISSUE #15149: autoReadDisabledRateLimiting is not corrent when limiting by pendingSendRequest
**Describe the bug**

in ServerCnx

```
   if (++pendingSendRequest == maxPendingSendRequests || isPublishRateExceeded) {
            // When the quota of pending send requests is reached, stop reading from socket to cause backpressure on
            // client connection, possibly shared between multiple producers
            ctx.channel().config().setAutoRead(false);
            recordRateLimitMetrics(producers);
            autoReadDisabledRateLimiting = isPublishRateExceeded;
            throttledConnections.inc();
        }
```

if ++pendingSendRequest == maxPendingSendRequests is true,  but isPublishRateExceeded is false.

autoReadDisabledRateLimiting = isPublishRateExceeded; will be false ,but it should be true

**Expected behavior**

autoReadDisabledRateLimiting = true;

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by leizhiyuan**
Fixes #15149 

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #15149 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

set autoReadDisabledRateLimiting true

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `no-need-doc` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Review by gaozhangmin — APPROVED** [review]
LGTM

**Review by gaozhangmin — APPROVED** [review]

**Comment by poorbarcode** [comment]
Hi @leizhiyuan I think `ServerCnx.autoReadDisabledRateLimiting` is for `broker or Netty-Thread`, not for `serverCnx`.  useful for that "When used to remove producer limiting, try to remove netty-thread limiting ( it will success when broker was not rate-limiting)". see: 

https://github.com/apache/pulsar/blob/732049fc6ca1beb046deb43057be2b130736fbca/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L2712-L2718

So I think this will break the original design.

We can have a friendly discussion.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 1]
```diff
@@ -2647,7 +2647,7 @@ public void startSendOperation(Producer producer, int msgSize, int numMessages)
             // client connection, possibly shared between multiple producers
             ctx.channel().config().setAutoRead(false);
             recordRateLimitMetrics(producers);
-            autoReadDisabledRateLimiting = isPublishRateExceeded;
+            autoReadDisabledRateLimiting = true;
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:2650]: This change does not follow the original design of `autoReadDisabledRateLimiting`.
Can you provide more info about the effect when `pendingSendRequest == maxPendingSendRequests` and `autoReadDisabledRateLimiting` is false.

**Review by Jason918 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java (hunk 1 above)
Comment (leizhiyuan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:2650]: ok，maybe I misunderstand the design

**Pull request closed without merging**